SofaKernel/modules/SofaBaseTopology/SofaBaseTopology_test/RegularGridTopology_test.cpp

```diff
@@ -207,20 +231,34 @@ std::vector<std::vector<int>> dimvalues={
     {-2,1,1, 1, 1},
 };
 
-TEST_P(RegularGridTopology_test, regularGridSize )
+TEST_P(RegularGridTopology_test, regularGridSizeComputeEdgeFromTriangle)
 {
     /// We check if this test should returns a warning.
-    if(GetParam()[3]==1){
-        {
-            EXPECT_MSG_EMIT(Warning) ;
-            ASSERT_TRUE( regularGridSize(GetParam()) );
-        }
-    }else{
-        ASSERT_TRUE( regularGridSize(GetParam()) );
+    if (GetParam()[3] == 1)
+    {
+        EXPECT_MSG_EMIT(Warning);
+        ASSERT_TRUE(regularGridSize(GetParam(), true));
+    }
+    else
+    {
+        ASSERT_TRUE(regularGridSize(GetParam(), true));
+    }
+}
+
+TEST_P(RegularGridTopology_test, regularGridSize)
+{
+    /// We check if this test should returns a warning.
+    if (GetParam()[3] == 1)
+    {
+        EXPECT_MSG_EMIT(Warning);
+        ASSERT_TRUE(regularGridSize(GetParam(), false));
```

Comment:
very good idea to put the computeTriangle as a parameter. IT would be even better to duplicate the test or at least have some with the option set to false (this will test the old behavior in fact)

Reply:
sorry I meant the opposite... set to true. To test the new behavior.

Reply:
sorry my bad, didn't see the lines above :+1:

Reply:
sorry my bad, didn't see the lines above 👍